### DISCUSSION TITLE
I want to make json with the help of page.extract_words and im.draw_rects

### DISCUSSION THREAD

**Discussion by Faizan-Saifullah** [jsvine/pdfplumber#546] [Q&A]
Hi
I have extracted words from a pdf page and i have been able to draw rectangles using im.draw_rects on those extracted words. Now i want to make json file that contains all this information of page.extract_words such that when i open in labelme , it shows same rects as im.draw_rects .

Here , is my code:

with pdfplumber.open(r'empty_test.pdf') as pdf:

    print(pdf.pages)
    page = pdf.pages[0]
    im = page.to_image(resolution = 400)
    text = page.extract_words()
    im = im.draw_rects(page.extract_words())
    im.save("result.jpeg")
    #h = open('empty_test' + '.json', "w")
    #json.dump(text, h, indent=2, sort_keys=False)
    #h.close()

    imm = im.annotated
    imm.save("result2.jpeg")
    print(page.extract_words())
    ann = im.original
    ann.save("scan.jpeg")

**Comment by samkit-jain** [comment]
Hi @Faizan-Saifullah Appreciate your interest in the library. To help you further, could you please share the JSON structure format in which you want to store the data?

**Reply by Faizan-Saifullah** [reply]
the structure is almost the same as the one being obtained from "print(page.extract_words())" :

[{'text': 'Requisition', 'x0': Decimal('217.800'), 'x1': Decimal('251.814'), 'top': Decimal('10.440'), 'bottom': Decimal('17.440'), 'upright': True, 'direction': 1}, {'text':  .....

Such that when I open in labelme , it displays the bounding boxes as well as extracted text.

**Reply by Faizan-Saifullah** [reply]
Above is Done!
Just one more thing (if you can please):
I am unable to convert bbox coordinates from pdf coords to image coords, so that the bbox for an "extracted_word" is displayed at correct position in the following file created in code: "ann.save("scan.jpeg")"

**Reply by samkit-jain** [reply]
You can use the protected method [`PageImage._reproject_bbox(...)`](https://github.com/jsvine/pdfplumber/blob/002500ab2e28885af1fa6e1b13c5234f874bcdae/pdfplumber/display.py#L78) to do that. It will give you the coordinates translated to the image. Example usage,

```python
import pdfplumber

pdf = pdfplumber.open("file.pdf")
page = pdf.pages[0]
image = page.to_image(resolution=200)

for word in page.extract_words():
    x0, top, x1, bottom = image._reproject_bbox((word["x0"], word["top"], word["x1"], word["bottom"]))
```
